modules/oe_whitelabel_search/src/Plugin/Block/SearchBlock.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_whitelabel_search\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\oe_whitelabel_search\Form\SearchForm;
+use Drupal\views\Entity\View;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Exposes a block with search box.
+ *
+ * @Block(
+ *  id = "whitelabel_search_block",
+ *  admin_label = @Translation("Whitelabel Search Block"),
+ *  category = @Translation("Blocks"),
+ * )
+ */
+class SearchBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The form builder service.
+   *
+   * @var \Drupal\Core\Form\FormBuilderInterface
+   */
+  protected $formBuilder;
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Construct CorporateEcLogoBlock object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\Core\Form\FormBuilderInterface $form_builder
+   *   The form builder service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory, FormBuilderInterface $form_builder) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $config_factory;
+    $this->formBuilder = $form_builder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('form_builder')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state): array {
+    $form = parent::blockForm($form, $form_state);
+    $config = $this->getConfiguration();
+
+    $form['form_action'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Form action'),
+      '#description' => $this->t('The url the form should submit to.'),
+      '#default_value' => $config['form']['action'],
+      '#required' => TRUE,
+    ];
+    $form['input_name'] = [
```

Comment:
Could be a good|great idea to group the fields of input and the fields of button grouped to clarify all.